### PR TITLE
Improve libtool.sh script

### PR DESCRIPTION
**Pull request by keith**
This is just some basic cleanup to pass shellcheck, and error if
toolchain setup didn't happen correctly.

### CODE REVIEW

File: tools/objc/libtool.sh [hunk 1]
```diff
@@ -96,7 +97,6 @@ while [[ $# -gt 0 ]]; do
      ARG="$1"
      shift
      ARGS+=("${ARG}")
-     OUTPUTFILE="${ARG}"
```
Comment (keith) [inline tools/objc/libtool.sh:99]: unused

File: tools/objc/libtool.sh [hunk 2]
```diff
@@ -120,7 +120,7 @@ while [[ $# -gt 0 ]]; do
      ;;
    # Remaining args are input objects
    *)
-     ARGS+=("$(echo "$(hash_objfile "${ARG}")")")
```
Comment (keith) [inline tools/objc/libtool.sh:123]: no need to echo the output from another call

File: tools/objc/libtool.sh [hunk 3]
```diff
@@ -44,7 +45,7 @@ elif "${MY_LOCATION}"/libtool_check_unique "$@"; then
 fi
 
 TEMPDIR="$(mktemp -d "${TMPDIR:-/tmp}/libtool.XXXXXXXX")"
-trap "rm -rf \"${TEMPDIR}\"" EXIT
```
Comment (keith) [inline tools/objc/libtool.sh:47]: this way it's evaluated later in case the value changes

File: tools/objc/libtool.sh [hunk 4]
```diff
@@ -35,7 +35,8 @@ fi
 WRAPPER="${MY_LOCATION}/xcrunwrapper.sh"
 
 if [ ! -f "${MY_LOCATION}"/libtool_check_unique ] ; then
-    echo "libtool_check_unique not found. Please file an issue at github.com/bazelbuild/bazel"
+  echo "libtool_check_unique not found. Please file an issue at github.com/bazelbuild/bazel"
+  exit 1
```
Comment (keith) [inline tools/objc/libtool.sh:39]: now that this script has exited for a while I think erroring if it doesn't exist is reasonable

**Comment by gregestren** [comment]
<puts on PR triaging hat> @keith do you know the best people to review PRs like this?

**Comment by keith** [comment]
I'll close this one, I ended up folding it into https://github.com/bazelbuild/bazel/pull/13953

**Pull request closed without merging**